pkg/persistence/result.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2019, 2020 the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
```

Comment:
for now I have copied this from restore/result.go.

Once there is agreement in general around this PR. I will potentially remove the restore/result.go and point to a common result.go file. 

This is small refactoring which can be done later

Reply:
putting in persistence is causing import loops.
SInce we need to import results in logging, it can either be in /internal or /util
Other places will cause import cycle